browser/js/lexicontool.factory.js

```diff
@@ -0,0 +1,9 @@
+app.factory('lexicon', function($http) {
+  return function(data) {
+    return $http.post('/api/lexicon/buffer', {words: data}).then(function(response) {
+      return $http.get('/api/lexicon/').then(function(secondResponse) {
+        return secondResponse.data;
+      });
+    });
+  }
```

Comment:
You can refactor this promise chain to be flat:

```js
app.factory('lexicon', function($http) {
  return function (data) {
    return $http.post('/api/lexicon/buffer', { words: data })
    .then(function (response) {
      return $http.get('/api/lexicon/')
    })
    .then(function (secondResponse) {
      return secondResponse.data;
    });
  };
});
```

More to the point — you're making two requests to your own server. Is there any reason why the first request can't just give you the response you want? @angelanjelly @hannahlorane